src/services/crawl-tx/chain.registry.ts

```diff
@@ -2,16 +2,23 @@
 import { Registry, TsProtoGeneratedType } from '@cosmjs/proto-signing';
 import { defaultRegistryTypes as defaultStargateTypes } from '@cosmjs/stargate';
 import { wasmTypes } from '@cosmjs/cosmwasm-stargate/build/modules';
-import { ibc, cosmos, aura } from '@aura-nw/aurajs';
 import { toBase64, fromUtf8, fromBase64, toUtf8 } from '@cosmjs/encoding';
 import { LoggerInstance } from 'moleculer';
 import _ from 'lodash';
 import { SemVer } from 'semver';
-import txRegistryType from './registry-type/aura-network.json' assert { type: 'json' };
-import { MSG_TYPE } from '../../common';
+import { ibc as auraIbc, cosmos as auraCosmos, aura } from '@aura-nw/aurajs';
+import {
```

Comment:
có 1 cách khác là import ở trong cái switch

Reply:
Là import hẳn trong chỗ switch case hả anh. Em thấy nó báo lỗi TS1232: An import declaration can only be used at the top level of a namespace or module.

Reply:
hmm sao search eslint ko ra cái rule này nhỉ

Reply:
em viết 
```
{ ibc, cosmos, seiprotocol } = await import('@horoscope-v2/sei-js-proto');
```
hay gì?

Reply:
Em import theo bình thường anh, kiểu như của anh vẫn báo lỗi ấy, em toàn import trên chứ chưa import dưới logic bao giờ

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html
đây này, hay thiếu cái esnext

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import

Reply:
Oke anh, em hiểu và code theo rùi, nhưng hiện tại constructor nó không cho await, không biết bỏ await thì có sao không nhưng thôi chắc em tách cái phần switch case sang một file riêng như lần trước anh bảo, đỡ viết trong registry thì sẽ dùng được await